core/domain/classifier_services.py

```diff
@@ -347,6 +348,75 @@ def mark_training_job_complete(job_id):
                                            feconf.TRAINING_JOB_STATUS_COMPLETE)
 
 
+def mark_training_job_failed(job_id):
+    """Updates the training job's status to failed.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_FAILED)
+
+
+def mark_training_job_pending(job_id):
+    """Updates the training job's status to pending.
+
+    Args:
+        job_id: str. ID of the ClassifierTrainingJob.
+    """
+    _update_classifier_training_job_status(job_id,
+                                           feconf.TRAINING_JOB_STATUS_PENDING)
+
+
+def update_failed_jobs(training_jobs):
+    """Updates status of job_models to failed.
+
+    Args:
+        training_jobs: list. List of jobs to be updated as failed.
+    """
+    for training_job in training_jobs:
+        mark_training_job_failed(training_job.job_id)
+
+
+def fetch_next_job():
+    """Gets next job model in the job queue.
+
+    Returns:
+        ClassifierTrainingJobModel. Domain object of the next training Job.
+    """
+    classifier_training_jobs = []
+    cursor = None
+    valid_jobs = []
+    failed_jobs = []
+
+    while len(valid_jobs) == 0:
```

Comment:
For (2), I have the name wrong. The point here is to make it such that the next poll _doesn't_ surface pending jobs which don't need to be checked since they are within the TTL, so TTL is the correct increment here rather than the polling frequency. So I agree with you that the name is misleading.

I'm having quite a bit of trouble coming up with a better name, though. Any suggestions? next_time_to_check, or something like that, perhaps?